src/ixdat/plotters/tpms_plotter.py

```diff
@@ -146,139 +141,452 @@ def plot_measurement(
+            x_unit (str): unit for x axis defaults to 's' for seconds
+            meta_units (dict): Dictionary with new units meta series to be plotted.
+                Default to ValueSeries.unit.name for last plotted ValueSeries
+            meta_list (list of str): The names of the data_series plotted on axes[1].
+                Defulats None.
+            T_name (str): The name of the value to plot on the right y-axis.
+                Defaults to the name of the series `measurement.temperature`
+            P_name (str): The name of the value to plot on the right y-axis.
+                Defaults to the name of the series `measurement.pressure`
+            T_color (str): Overwrite standard color to plot the variable given by T_name
+            P_color (str): Overwrite standard color to plot the variable given by P_name
+            logplot (bool): Whether to plot the MS data on a log scale (default True
+                unless mass_lists are given)
+            logdata (bool): Whether to plot the MS data on a log scale (default True
+                unless mass_lists are given)
+            legend (bool): Whether to use a legend for the MS data (default True)
+            emphasis (str or None): "top" for bigger top panel, "bottom" for bigger
+                bottom panel, None for equal-sized panels, "one figure" to plot all in
+                one figure
+            kwargs (dict): Additional kwargs go to all calls of matplotlib's plot()
+
+        Returns:
+                axes[0] is MS data;
+                axes[1] is meta data (default temperature and pressure);
+        """
+
+        measurement = measurement or self.measurement
 
-        axes = self.plot_measurement(
+        if not ax:
+            ax = (
+                axes[0]
+                if axes
+                else self.new_ax(ylabel=f"signal / [{unit}]", xlabel="time / [s]")
+            )
+        axes = axes or [ax, ax.twinx()]  # prepare an axis unless we were given two.
+
+        # then we have MS data!
+        self.ms_plotter.plot_measurement(
             measurement=measurement,
-            axes=axes,
-            mass_list=mass_list,
-            mass_lists=mass_lists,
-            mol_list=mol_list,
-            mol_lists=mol_lists,
+            ax=axes[0],
             tspan=tspan,
             tspan_bg=tspan_bg,
             remove_background=remove_background,
+            mass_list=mass_list,
+            mol_list=mol_list,
             unit=unit,
             x_unit=x_unit,
-            T_name=T_name,
-            T_color=T_color,
             logplot=logplot,
             logdata=logdata,
             legend=legend,
-            emphasis=emphasis,
             **kwargs,
         )
 
+        T_name = T_name or measurement.T_name
+        P_name = P_name or measurement.P_name
+
+        # figure out if one ot two axes is to be plotted in bottom panel
+        if not meta_list:
```

Comment:
bleh, okay, if you need to do the whole thing with *meta_list(s)* again, maybe I was wrong and it *is* more elegant to combine some of the code again. But again, I think it can be done simpler.

Reply:
This might simply be my limited knowledge of python. But i see that I either combine the dataseries in a meta_list or TP_list or duplicate ~20 lines of code to plot with correct labels, colours, and units ? 

the list is now called TP list but effectively the codebase is the same :/

Reply:
I'll give this a try after we merge. Will need my own fingers on it to understand why the complexity is needed.

Reply:
TP_list is better than meta_list :) 
You could also define a function and call it twice to avoid duplicating code.